### PR TITLE
[FEATURE] Use optgroups for CType field

### PR DESCRIPTION
**Pull request by achimfritz**

### CODE REVIEW

**Review by georgringer — APPROVED** [review]

File: Classes/Tca/Registry.php [hunk 1]
```diff
@@ -30,7 +29,12 @@ public function configureContainer(ContainerConfiguration $containerConfiguratio
         ExtensionManagementUtility::addTcaSelectItem(
             'tt_content',
             'CType',
-            [$containerConfiguration->getLabel(), $containerConfiguration->getCType(), $containerConfiguration->getCType()]
+            [
+                $containerConfiguration->getLabel(),
+                $containerConfiguration->getCType(),
+                $containerConfiguration->getCType(),
+                $containerConfiguration->isUseContainerOptgroup() ? 'container' : ''
```
Comment (bmack) [inline Classes/Tca/Registry.php:36]: wouldn't it be cool if one could add their own optgroup?

I'd use EXT:container extensively, and then group by "2 columns", "3 columns" etc. and have that as group label directly?

Comment (achimfritz) [reply Classes/Tca/Registry.php:36]: IMHO you should group them in the "New Content Element Wizard" (this is our prefered Workflow). This is done by setting  the "registerInNewContentElementWizard" in Container-Registration to false and write your own NewContentElementWizard-Configuration.
If one can add their own optgroup, 2 parameters are required, the key, and the label. I think this is more confusing than helpful.

Comment (achimfritz) [reply Classes/Tca/Registry.php:36]: i will change this to Bennis suggestion

Comment (achimfritz) [reply Classes/Tca/Registry.php]: changed, and: the `$group` is also used in New Content Element Wizard as Tab

**Review by bmack — APPROVED** [review]

**Review by ervaude — APPROVED** [review]

**Pull request merged**